projects/lexical_plugin/lib/lexical/plugin/diagnostic/result.ex

```diff
@@ -1,22 +1,40 @@
 defmodule Lexical.Plugin.Diagnostic.Result do
-  defstruct [:location, :message, :severity, :source, :uri]
+  alias Lexical.Document
+  defstruct [:details, :message, :position, :severity, :source, :uri]
 
-  @type severity :: :warning | :error
+  @type path_or_uri :: Lexical.path() | Lexical.uri()
+  @type severity :: :message | :warning | :error
+  @type mix_position ::
+          non_neg_integer()
+          | {pos_integer(), non_neg_integer()}
+          | {pos_integer(), non_neg_integer(), pos_integer(), non_neg_integer()}
+
+  @type position :: mix_position() | Document.Range.t() | Document.Position.t()
   @type t :: %__MODULE__{
-          location: Lexical.Document.Range.t() | Lexical.Document.Position.t(),
+          position: position,
```

Comment:
Is there any consideration for using "position" instead of "location" here?

Reply:
Yeah, I felt that despite elixir calling them locations, I went with the lsp terminology. They're very close in meaning.